### ISSUE TITLE
LilyPond new static binaries

### ISSUE THREAD

**Issue by fedelibre** [MutopiaProject/mupub#3]
Since version 2.23.6 LilyPond binaries are available on Gitlab releases as static installation (architecture specific).
It should be easier to use a LilyPond binary (you only have to unpack the tarball and copy the files), but mupub should be updated accordingly.

Here's the new download_url (no fallback):
https://gitlab.com/lilypond/lilypond/-/releases/

**Comment by fedelibre** [comment]
@glenl Can I tell mupub to use a local binary instead of downloading from Internet?

I'm stuck here:

```
$ mupub check
INFO     mupub.lily   Compiler installation needed for 2.24.0
INFO     mupub.lily   Trying http://download.linuxaudio.org/lilypond/binaries/
INFO     mupub.lily   Trying http://lilypond.org/downloads/binaries/
WARNING  mupub.lily   No install scripts found for 2.24.0
Failed to determine (or install) compiler.
```

**Comment by fedelibre** [comment]
To move on I tricked mupub and created a directory `~/.mupub/lycache/2.24.3` where I copied the bin/ and the other directories from the lilypond installation.

I see that mupub wants to use the exact version of lilypond defined in the source file. Let's say I update a lilypond file using convert-ly: right now convert-ly 2.24.3 will add the version 2.24.0. But if I want mupub to use the latest stable (2.24.3) I should put that exact version in the source file. Is this the recommended way for Mutopia?

**Comment by glenl** [comment]
The purpose of mupub was to have a repeatable process so, yes, the exact version should be in the source file. Should that "rule" be loosened? If a file states version 2.24.0 should it just find a 2.24.* compiler in the cache and move on?

**Comment by fedelibre** [comment]
Yes, I agree that the rule may be loosened as you say. Possibly using the latest version in 2.24.*.
